parseSchema/index.js

```diff
@@ -0,0 +1,78 @@
+"use strict";
+
+const Parse = require('parse/node');
+
+function updateSchemaOnlyWhenNecessary(existingSchemaData, intendedSchema)
+{
+  // remove any existing fields from the request.
+  Object.getOwnPropertyNames(existingSchemaData.fields).forEach(existingField => { delete intendedSchema._fields[existingField] ; });
+  return (Object.getOwnPropertyNames(intendedSchema._fields).length + Object.getOwnPropertyNames(intendedSchema._indexes).length > 0 ) ? 
+    intendedSchema.update() : Promise.resolve(existingSchemaData);
+}
+
+function setParseSchema (serverConfig) {
+
+  Parse.initialize(serverConfig.appId, null, serverConfig.masterKey);
+  Parse.serverURL = serverConfig.serverURL ;
+  
+/*
+  Notes:
+
+  (1) A Parse.Schema only represents data needed to make a request about a Parse Schema.  A better name would be SchemaRequest.
+  
+  (2) As such, fields added by Parse.Schema.prototype.addField() function and its ilks only represent intention to add/create them, and nothing else.
+      For example, the .get() method doesn't change any such "fields" in the Schema object based on the result returned from the Parse server.
+  
+  (3) Starting from Parse v2, standard Javascript Promise has replaced Parse.Promise, which unfortunately is still used in sample code
+      in ParsePlatform documentation as of this writing.
+ */
+  const contactSchema = new Parse.Schema('contacts');
+
+  contactSchema
+    .addString('birthCountry')
+    .addString('ethnicity')
+    .addString('race')
+    .addString('genderIdentity')
+    .addString('uid')
+    .addDate('dateOfBirth')
+    .addDate('firstInjectionAge')
+    .addDate('dateOfBirth')
+    .addBoolean('hispanic')
```

Comment:
Do these get automatically created?
- `updatedAt`
- `createdAt`
- `ACL`

Can you ensure these are added?
- `primaryDrug`
- `hepCStatus`
- `hasHealthInsurance`
- `didOdLastYear`
- `hivStatus`
- `isEnrolled`
- `dateOfLastVisit`
- `countryOfBirth`
- `isInCareForHepC`
- `ageOfFirstInjection`
- `isInCareForHiv`
- `housingStatus`
- `otherDrugsAggregate`
- `profileNotes`
- `otherDrugs`
- `syringesGivenAggregate`
- `zipCode`
- `syringesTakenAggregate`
- `healthInsurer`

Please remove:
- `birthCountry` as it's replaced with `countryOfBirth`
- `firstInjectionAge` as it's replaced by `ageOfFirstInjection`

Reply:
ok...  will do that.  I copied them verbatim from the front-end repo, which must have been outdated.

Reply:
yes, ACL, updatedAt and createAt are built-in and no need to add.